RedisEnterprise-DNS-Migration/migration_script.ps1

```diff
@@ -190,53 +190,57 @@ function Migrate-Vnets($PrivateLinkMap, $PrivateDnsZonesMap) {
-                            $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' has an IP address of $($peIpAddresses) which is not contained in the private DNS zones list of A records which include [$($zoneARecordIpAddresses -join ", ")]? Do you want to and add a new one dns zone group to fix this?"
-                        }
+                        $shouldProceed = ShouldRemovePrivateDnsZoneGroup $existingGroup $privateLink $newZone
                         if ($shouldProceed) {
-                            Write-Host "Removing old Dns Zone Group (A-Record) for private link '$($privateLink.Name)'" -ForegroundColor Yellow
+                            Write-Host "Removing old DNS zone group (A record) from private endpoint '$($privateLink.Name)'" -ForegroundColor Yellow
                             Remove-AzPrivateDnsZoneGroup -PrivateEndpointName $privateLink.Name -Name $existingGroup.Name -ResourceGroupName $privateLink.ResourceGroupName -Force
                         } 
                     } else {
                         
                         Write-Host "[Step 4 under section 'How to migrate to the new private DNS zone' in migration document]" -ForegroundColor Yellow
-                        $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' has no associated DNS zone group. These groups automatically manage the A-Records creation and deletion for the private endpoint. Do you want to create a new one?"
+                        $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' has no associated Dns zone group. These groups automatically manage the A records creation and deletion for the private endpoint. Do you want to create a new one?"
                     }
                     if($shouldProceed) {
-                            Write-Host "Creating new Dns Zone Group (A-Record) for private link '$($privateLink.Name)'" -ForegroundColor Yellow
+                            Write-Host "Creating new DNS zone group (A record) for private endpoint '$($privateLink.Name)'" -ForegroundColor Yellow
                             New-AzPrivateDnsZoneGroup -ResourceGroupName $privateLink.ResourceGroupName -PrivateEndpointName $privateLink.Name -Name ($privateLink.Name + "-" + $privateDnsZoneConfig.Name) -PrivateDnsZoneConfig $privateDnsZoneConfig
                     }
                 }
             }
-            Write-Host "Finished migration process for VNET: $($vnet.Name)." -ForegroundColor Green
+            Write-Host "Finished migration process for VNet: $($vnet.Name)." -ForegroundColor Green
         }
     }
     Write-Host "Finished configuring VNets with correct private DNS zones." -ForegroundColor Green
 }
 
+function ShouldRemovePrivateDnsZoneGroup($existingGroup, $privateLink, $newZone) {
+    if (!$shouldProceed -and $null -eq $existingGroup.PrivateDnsZoneConfigs) {
```

Comment:
Not that it will cause any problems, but just for readability for any users who skim the code, I think the `$shouldProceed` check can be removed now that these checks are in their own function?

```suggestion
    if ($null -eq $existingGroup.PrivateDnsZoneConfigs) {
```